hazelcast/src/test/java/com/hazelcast/jet/core/JobUploadFailureTest.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.instance.impl.HazelcastBootstrap;
+import com.hazelcast.jet.JetException;
+import com.hazelcast.jet.JetService;
+import com.hazelcast.jet.SubmitJobParameters;
+import com.hazelcast.jet.config.JetConfig;
+import com.hazelcast.jet.impl.JetClientInstanceImpl;
+import com.hazelcast.jet.test.SerialTest;
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.FileSystemException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.when;
+
+@Category({SerialTest.class})
+public class JobUploadFailureTest extends JetTestSupport {
```

Comment:
I think it's a good idea to add a test in which jar is copied as a normal (non-classpath) file, uploaded and we check that after failure in the job, the jar is cleaned up

Reply:
Add a new test_jar_isDeleted test to check that the jar is deleted after a failure